### PR TITLE
Sign In Redirect

### PR DESCRIPTION
**Pull request by calebjacob**
Reworked sign in redirect logic to use local storage instead of a query param since the query param didn't support the wallet modal redirect flow.

### CODE REVIEW

File: src/components/vm/VmInitializer.tsx [hunk 1]
```diff
@@ -96,21 +98,21 @@ export default function VmInitializer() {
...
-  );
+  const requestSignIn = useCallback(() => {
+    signInRedirect.saveCurrentUrl();
+    router.push('/signin');
+  }, [router, signInRedirect]);
+
+  const requestCreateAccount = useCallback(() => {
+    signInRedirect.saveCurrentUrl();
+    router.push('/signup');
+  }, [router, signInRedirect]);
```
Comment (shelegdmitriy) [inline src/components/vm/VmInitializer.tsx:115]: But this two functions are the same but with different names why do we need both?

Comment (shelegdmitriy) [inline src/components/vm/VmInitializer.tsx:115]: How about this
```suggestion
  const requestAuthentication = useCallback(() => (redirect) =>{
    signInRedirect.saveCurrentUrl();
    router.push(redirect);
  }, [router, signInRedirect]);
```
?
I mean this two methods are almost the same so why do we them separately?

**Review by marcinbodnar — COMMENTED** [review]
This functionality can be a separate logic, without changes to components. Instead of changing components, it can be closed into one function that will handle it.

For example, we can prepare a function that will detect if a user is opening `signin` or `signup` route and store the previous route in local storage, the same function can detect if the user is going back to near.org from auth process, get the route data from local storage and redirect to the previous route after the auth process is finished.

Thanks to that the codebase will be simpler, and since it will be a separate functionality, we don't need to think about this part when introducing further changes.

File: src/components/vm/VmInitializer.tsx (hunk 1 above)
Comment (calebjacob) [reply src/components/vm/VmInitializer.tsx]: Good point. I refactored these two methods into a single method provided by the `useSignInRedirect()` hook. Take a look and let me know what you think.

**Comment by calebjacob** [comment]
@marcinbodnar Thanks for the comment! I pulled out the `requestSignIn` and `requestCreateAccount` methods from the `VmInitializer` flow and now have a simple method provided by the new redirect hook. I think this is an improvement over what we currently have. This can always be refactored further down the line, but right now it's probably more important to make sure the redirect is working for our users.

**Comment by shelegdmitriy** [comment]
@calebjacob preview env is broken

**Comment by shelegdmitriy** [comment]
Also can't run it locally:
Знімок екрана 2023-06-26 о 20 37 58

**Comment by calebjacob** [comment]
@shelegdmitriy Thanks - should be fixed now.

**Review by shelegdmitriy — APPROVED** [review]
LGTM! This looks much cleaner 🙂

**Pull request merged**